### PR TITLE
#1592 Issue: Moved restore from archive to archived goal circle action

### LINKED ISSUE #1592: Move restore from archive to archived goal circle actions
**Is your feature request related to a problem? Please describe.**
It's very annoying when you want to restore something from archive and the + button is covering the restore from archive button.

**Describe the solution you'd like**
Allow a menu to appear when pressing the archived goal circle. In the menu you have two choices: 'Restore' and 'Delete forever'.

**Describe alternatives you've considered**
Just move the reatore from archive to the left. Then there is no way of deleting the archived/completed goal - and the interface is less consistent.

**Additional context**

### PR DESCRIPTION
**Pull request by siva9495**
Resolves #1592 
- Moved Restore from archive to archived goal circle.
- Checked darkmode also.

### CODE REVIEW

**Comment by tijlleenders** [comment]
Very nice @siva9495 !

@Tushar-4781 Any comments on the code?

**Comment by tijlleenders** [comment]
Nicely done @siva9495 !

@Tushar-4781 OK to merge?

File: src/components/GoalsComponents/MyGoalActions/MyGoalActions.tsx [hunk 1]
```diff
@@ -102,52 +109,88 @@ const MyGoalActions = ({ goal, open }: { open: boolean; goal: GoalItem }) => {
...
+              className="goal-action-archive shareOptions-btn"
+              onClickCapture={async (e) => {
+                e.stopPropagation();
+                await handleActionClick("restore");
+              }}
+            >
+              <ActionDiv
+                label={t("Restore")}
+                icon={
+                  <img
```
Comment (Tushar-4781) [inline src/components/GoalsComponents/MyGoalActions/MyGoalActions.tsx:124]: pass the string not the icon. Checkout the icon component in common folder

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
@siva9495 can you please look at the changes

File: src/components/GoalsComponents/MyGoalActions/MyGoalActions.tsx (hunk 1 above)
Comment (siva9495) [reply src/components/GoalsComponents/MyGoalActions/MyGoalActions.tsx:124]: In the common folder there is no unarchieved icon. previously, also used same like this for the unarchived icon. So, that I pass the icon here also. can you please check this again.
Screenshot 2023-12-10 161348

File: src/components/GoalsComponents/MyGoalActions/MyGoalActions.scss [hunk 2]
```diff
@@ -25,6 +25,18 @@
   .goal-action:last-child {
     border-radius: 0 0 28px 0 !important;
   }
+
+  .goal-action-archive {
```
[the remark this answers is not in the document]
Comment (siva9495) [reply src/components/GoalsComponents/MyGoalActions/MyGoalActions.scss:29]: If I use those goal-action attributes for archive and delete, I notice a different design modal.
Screenshot 2023-12-10 163915

[the remark this answers is not in the document]
Comment (siva9495) [reply src/components/GoalsComponents/MyGoalActions/MyGoalActions.scss:29]: For this type of modal design, I created a new attributes with slight differences.
Screenshot 2023-12-10 165252

**Comment by tijlleenders** [comment]
@Tushar-4781 I think @siva9495 replied to the requested changes.

File: src/components/GoalsComponents/MyGoalActions/MyGoalActions.tsx (hunk 1 above)
Comment (Tushar-4781) [reply src/components/GoalsComponents/MyGoalActions/MyGoalActions.tsx:124]: i see. no prob. thanks for trying we will resolve this in near future

**Review by Tushar-4781 — DISMISSED** [review]
@siva9495 we will resolve the extra code and icon is in near future. let's get this PR through. Please resolve the conflicts to get it merged

**Review by Tushar-4781 — COMMENTED** [review]
Wow!! that's reverting some old progress. @vinaybadgujar102 can you please check this PR i think you splitted the MyGoal component into pieces right ?

**Comment by tijlleenders** [comment]
> @siva9495 we will resolve the extra code and icon is in near future. let's get this PR through. Please resolve the conflicts to get it merged

> Wow!! that's reverting some old progress. @vinaybadgujar102 can you please check this PR i think you splitted the MyGoal component into pieces right ?

@Tushar-4781 I don't understand the progression of the last two comments. @siva9495 just had to resolve merge conflicts and he did...  
What is still required to get this PR in now?

**Comment by tijlleenders** [comment]
@siva9495 Do you need help resolving the merge conflicts? It can be tricky sometimes...

**Comment by siva9495** [comment]
> @siva9495 Do you need help resolving the merge conflicts? It can be tricky sometimes...

yes need help.

**Comment by Tushar-4781** [comment]
> > @siva9495 we will resolve the extra code and icon is in near future. let's get this PR through. Please resolve the conflicts to get it merged
> 
> > Wow!! that's reverting some old progress. @vinaybadgujar102 can you please check this PR i think you splitted the MyGoal component into pieces right ?
> 
> @Tushar-4781 I don't understand the progression of the last two comments. @siva9495 just had to resolve merge conflicts and he did...  
> What is still required to get this PR in now?

He did resolve the merge conflict but in that process he reverted some work of @vinaybadgujar102 . I think he can explain better.

**Comment by siva9495** [comment]
> > > @siva9495 we will resolve the extra code and icon is in near future. let's get this PR through. Please resolve the conflicts to get it merged
> > 
> > 
> > > Wow!! that's reverting some old progress. @vinaybadgujar102 can you please check this PR i think you splitted the MyGoal component into pieces right ?
> > 
> > 
> > @Tushar-4781 I don't understand the progression of the last two comments. @siva9495 just had to resolve merge conflicts and he did...
> > What is still required to get this PR in now?
> 
> He did resolve the merge conflict but in that process he reverted some work of @vinaybadgujar102 . I think he can explain better.

@Tushar-4781 I do not know how some other file changes are coming into my changes. like I have no idea about those changes. please help me out with this issue.

**Comment by Tushar-4781** [comment]
@siva9495 for now just create a new pr :)

**Comment by tijlleenders** [comment]
@siva9495 No worries. We'll look into this. Probably we need to add some explanations to the contributor docs about merging with conflicts too.  
Thanks for being transparent and pointing out your issue!

**Comment by siva9495** [comment]
> @siva9495 for now just create a new pr :)

ok i will create new PR.

**Pull request closed without merging**